packages/peregrine/lib/hooks/usePagination.js

```diff
@@ -73,6 +73,12 @@ export const usePagination = (props = {}) => {
         [history, location, searchParam]
     );
 
+    useMemo(() => {
+        if (initialPage && initialPage !== currentPage) {
+            setCurrentPage(initialPage);
+        }
+    }, [initialPage, currentPage]);
```

Comment:
For starters let's use `useEffect` instead of `useMemo`. While memoization has a side effect of invoking the callback, the effect hook is more appropriate here. And because `initialPage` is always truthy (it has a default value fallback) you can skip the truthy check.

```
    useEffect(() => {
        if (initialPage !== currentPage) {
            setCurrentPage(initialPage);
        }
    }, [currentPage, initialPage]);
```

I was wondering though why we would want to call `setCurrentPage` directly and not use the `setPage` callback.

Reply:
I have just updated code use `useEffect`. I have to use directly `setCurrentPage ` cause your `setPage` callback will reset param on URL, that mean when you back from page 2 to page 1 then you could not click next to page 2 although page 2 data was loaded.